.github/workflows/wheels.yml

```
@@ -190,17 +190,28 @@ jobs:
 
     - run: make sdist
 
-    - uses: actions/upload-artifact@v3
+    - uses: actions/upload-artifact@v4
       with:
-        name: dist
+        name: dist-sdist
         path: dist/*.tar.gz
 
-  success:
-    permissions:
-      contents: none
+  pypi-publish:
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
```

Let's check it can upload to TestPyPI:
```suggestion
    # TEMP for testing
    # if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
    if: github.event_name == 'push'
```